analyzer/consensus/consensus.go

```diff
@@ -66,27 +67,29 @@ func OpenSignedTxNoVerify(signedTx *transaction.SignedTransaction) (*transaction
 
 // processor is the block processor for the consensus layer.
 type processor struct {
-	mode    analyzer.BlockAnalysisMode
-	history config.History
-	source  nodeapi.ConsensusApiLite
-	network sdkConfig.Network
-	target  storage.TargetStorage
-	logger  *log.Logger
-	metrics metrics.AnalysisMetrics
+	chainName common.ChainName
+	mode      analyzer.BlockAnalysisMode
+	history   config.History
+	source    nodeapi.ConsensusApiLite
+	network   sdkConfig.Network
+	target    storage.TargetStorage
+	logger    *log.Logger
+	metrics   metrics.AnalysisMetrics
 }
 
 var _ block.BlockProcessor = (*processor)(nil)
 
 // NewAnalyzer returns a new analyzer for the consensus layer.
-func NewAnalyzer(blockRange config.BlockRange, batchSize uint64, mode analyzer.BlockAnalysisMode, history config.History, source nodeapi.ConsensusApiLite, network sdkConfig.Network, target storage.TargetStorage, logger *log.Logger) (analyzer.Analyzer, error) {
+func NewAnalyzer(chainName common.ChainName, blockRange config.BlockRange, batchSize uint64, mode analyzer.BlockAnalysisMode, history config.History, source nodeapi.ConsensusApiLite, network sdkConfig.Network, target storage.TargetStorage, logger *log.Logger) (analyzer.Analyzer, error) {
 	processor := &processor{
-		mode:    mode,
-		history: history,
-		source:  source,
-		network: network,
-		target:  target,
-		logger:  logger.With("analyzer", consensusAnalyzerName),
-		metrics: metrics.NewDefaultAnalysisMetrics(consensusAnalyzerName),
+		chainName: chainName,
```

Comment:
take this from a field in history instead of pulling chainName from the config. this is so that we can unset chainName and use a custom history if we need to adjust something (i.e. something unrelated to the static accounts set) at runtime.

having a "mainnet" and "testnet" known static account sets is fine, and the new history field can take on the values "mainnet" and "testnet."

Reply:
updated, check again please.

Reply:
lg thanks